src/main/java/org/terasology/books/logic/BookcaseSystem.java

```diff
@@ -57,18 +57,15 @@ public void filterBook(BeforeItemPutInInventory event, EntityRef entity) {
      * @param event the triggering event (something caused block drops - the bookcase was destroyed)
      * @param entity the bookcase's entity, has the target inventory to go through
      */
-    @ReceiveEvent(components = {BookcaseComponent.class})
-    public void onDestroyBookCase(CreateBlockDropsEvent event, EntityRef entity) {
-        if (entity.hasComponent(InventoryComponent.class)) {
-            Vector3f location = entity.getComponent(LocationComponent.class).getWorldPosition();
-            InventoryComponent inventory = entity.getComponent(InventoryComponent.class);
-            List<EntityRef> items = inventory.itemSlots;
-            for (EntityRef item: items) {
-                if (item.hasComponent(BookComponent.class)) {
-                    item.send(new DropItemEvent(location));
-                }
+    @ReceiveEvent(components = {BookcaseComponent.class, InventoryComponent.class, LocationComponent.class})
+    public void onDestroyBookCase(CreateBlockDropsEvent event, LocationComponent location, InventoryComponent inventory, EntityRef entity) {
```

Comment:
It is enough to have it either in the annotation (if you want to check that the component is there, but don't need it for computation) or as argument (if you want to access or modify the component). 
Also, I'm not sure whether the argument order matters or not, but usually we go for event, entity, components.

```suggestion
    @ReceiveEvent(components = {BookcaseComponent.class})
    public void onDestroyBookCase(CreateBlockDropsEvent event, EntityRef entity, LocationComponent location, InventoryComponent inventory) {
```